compliance-api/src/compliance_api/services/cached_staff_user.py

```diff
@@ -0,0 +1,78 @@
+"""Cached service for staff user validation during token authentication."""
+
+from typing import Optional
+
+from flask import current_app
+
+from compliance_api.models.staff_user import StaffUser as StaffUserModel
+from compliance_api.utils.cache import cache
+
+
+class CachedStaffUserService:
+    """Cached staff user service for performance optimization during token validation."""
+
+    CACHE_TIMEOUT = 3600  # 1 hour
+    STAFF_CACHE_KEY_PREFIX = "staff_user:"
+    ALL_STAFF_CACHE_KEY = "all_staff_users"
+
+    @classmethod
+    def get_staff_by_auth_guid(cls, auth_guid: str) -> Optional[StaffUserModel]:
+        """
+        Get staff user by auth_guid with caching.
+
+        This method is optimized for token validation performance.
+        It first checks cache, then falls back to database if not found.
+
+        Args:
+            auth_guid: The auth_user_guid from token (preferred_username)
+
+        Returns:
+            StaffUser model instance if found and active, None otherwise
+        """
+        if not auth_guid:
+            return None
+
+        cache_key = f"{cls.STAFF_CACHE_KEY_PREFIX}{auth_guid}"
+
+        # Try to get from cache first
+        cached_staff = cache.get(cache_key)
+        if cached_staff is not None:
+            current_app.logger.debug(f"Cache hit for staff user: {auth_guid}")
+            # Return None if cached value is explicitly False (user doesn't exist)
+            return cached_staff if cached_staff else None
+
```

Comment:
i think the return method doesnt have to check for none since its already inside the if check